processing/src/main/java/org/apache/druid/query/groupby/GroupByQueryConfig.java

```diff
@@ -43,12 +48,14 @@
   private static final String CTX_KEY_BUFFER_GROUPER_MAX_LOAD_FACTOR = "bufferGrouperMaxLoadFactor";
   private static final String CTX_KEY_BUFFER_GROUPER_MAX_SIZE = "bufferGrouperMaxSize";
   private static final String CTX_KEY_MAX_ON_DISK_STORAGE = "maxOnDiskStorage";
-  private static final String CTX_KEY_MAX_SELECTOR_DICTIONARY_SIZE = "maxSelectorDictionarySize";
-  private static final String CTX_KEY_MAX_MERGING_DICTIONARY_SIZE = "maxMergingDictionarySize";
   private static final String CTX_KEY_FORCE_HASH_AGGREGATION = "forceHashAggregation";
   private static final String CTX_KEY_INTERMEDIATE_COMBINE_DEGREE = "intermediateCombineDegree";
   private static final String CTX_KEY_NUM_PARALLEL_COMBINE_THREADS = "numParallelCombineThreads";
   private static final String CTX_KEY_MERGE_THREAD_LOCAL = "mergeThreadLocal";
+  private static final double MERGING_DICTIONARY_HEAP_FRACTION = 0.3;
+  private static final double SELECTOR_DICTIONARY_HEAP_FRACTION = 0.10;
+  private static final long MIN_AUTOMATIC_DICTIONARY_SIZE = 1_000_000;
+  private static final long MAX_AUTOMATIC_DICTIONARY_SIZE = 1_000_000_000;
```

Comment:
These two MIN and MAX ensure the range of automatic computed value, I think it's better to document them.

Reply:
BTW, these two are in decimal byte format, should we use binary byte which is 1024*1024 and 1024*1024*1024 ?

Reply:
Yeah, I agree. I changed MIN to 1 byte and didn't document it (it seems unnecessary to mention). I documented the 1GB max.

Reply:
How about defining the MAX as 1GiB instead of 1GB? I think 1GiB would be less confusing.

Reply:
Ah, I missed the comment about binary bytes. I dunno if there is a reason to use binary bytes. I kept it decimal since they both seem fine to me.

Reply:
Hmm. Why is it less confusing to use 1GiB instead of 1GB? The old default (100MB) was in decimal.

Reply:
When representing a size of memory, binary format is commonly used while decimal format is usually for disk size. 
It's OK to keep it in decimal format.

Reply:
Ah, I see. I wrote it as decimal since I am not familiar with that convention. I'll leave it as is for this patch.